src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj

```diff
@@ -8,6 +8,9 @@
     <NoWarn>$(NoWarn);SYSLIB1103,SYSLIB1104</NoWarn>
     <InterceptorsPreviewNamespaces>$(InterceptorsPreviewNamespaces);Microsoft.Extensions.Configuration.Binder.SourceGeneration</InterceptorsPreviewNamespaces>
     <EnableConfigurationBindingGenerator>true</EnableConfigurationBindingGenerator>
+
+    <!-- Avoid conflict with Microsoft.Extensions.Configuration.Binder (via InternalsVisibleTo) -->
+    <NoIncludeNullableAttributes>true</NoIncludeNullableAttributes>
```

Comment:
Is this still needed? If I understand correctly, you shouldn't get any conflicts now as we are not injecting any code for the nullable attribute definition in the source gen test.

Reply:
> Is this still needed?

It is. From https://github.com/dotnet/runtime/pull/101336#discussion_r1575655302:

> `Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests.csproj` has its own `NullableAttributes.cs` that it has ignored, but when actually used it conflicts with `internal` one in `Microsoft.Extensions.Configuration.Binder` [via `[InternalsVisibleTo]`].

Reply:
The attribute definition is inserted to `Microsoft.Extensions.Configuration.Binder` only with the condition `'$(TargetFrameworkIdentifier)' != '.NETCoreApp'"`.  If the test target is `.NETCoreApp`, then you shouldn't get any conflict because the attribute will not be included in `Microsoft.Extensions.Configuration.Binder`. 
If the test target is netfx, then the test itself will not have the definition of the attribute and I guess shouldn't get a conflict too because the attribute will be defined only in  `Microsoft.Extensions.Configuration.Binder`. did you try to remove `NoIncludeNullableAttributes` and still getting the conflict error?

Reply:
I pushed a revert to demonstrate the failure: https://github.com/dotnet/runtime/pull/101336/checks?check_run_id=24337379290.

> artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Release\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(245,58): error CS0436: (NETCORE_ENGINEERING_TELEMETRY=Build) The type 'NotNullIfNotNullAttribute' in 'D:\a\_work\1\s\src\libraries\System.Private.CoreLib\src\System\Diagnostics\CodeAnalysis\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'Microsoft.Extensions.Configuration.Binder, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in 'D:\a\_work\1\s\src\libraries\System.Private.CoreLib\src\System\Diagnostics\CodeAnalysis\NullableAttributes.cs'.

The failure is attributed to lines 225 and 245 of `BindingExtensions.g.cs`, where `[NotNullIfNotNull]` is used. The duplicate `NotNullIfNotNullAttribute` is not actually new (it's been inserted into both `net462` assemblies by `Directory.Build.targets` all along), but it only manifests as CS0436 when it comes time to decide which attribute to use.

So either we can prevent the duplicate (by letting the test project opt out of the include, in favor of the one from MECB), or just ignore CS0436 in the SG test project. The former seems preferable to me.

Reply:
Sorry for convenience. 

I am seeing after you reverted the `NoIncludeNullableAttributes` we still getting the failure. 

```
##[error]artifacts\obj\Microsoft.Extensions.Configuration.Binder.SourceGeneration.Tests\Release\net462\generated\Microsoft.Extensions.Configuration.Binder.SourceGeneration\Microsoft.Extensions.Configuration.Binder.SourceGeneration.ConfigurationBindingGenerator\BindingExtensions.g.cs(225,58): error CS0436: (NETCORE_ENGINEERING_TELEMETRY=Build) The type 'NotNullIfNotNullAttribute' in 'D:\a\_work\1\s\src\libraries\System.Private.CoreLib\src\System\Diagnostics\CodeAnalysis\NullableAttributes.cs' conflicts with the imported type 'NotNullIfNotNullAttribute' in 'Microsoft.Extensions.Configuration.Binder, Version=9.0.0.0, Culture=neutral, PublicKeyToken=adb9793829ddae60'. Using the type defined in 'D:\a\_work\1\s\src\libraries\System.Private.CoreLib\src\System\Diagnostics\CodeAnalysis\NullableAttributes.cs'.
D:\a\_work\1\s\artifa
```

Please return back `NoIncludeNullableAttributes`. Thanks!

Reply:
Looks you need to rebase with the main one more time.

Reply:
My reply crossed yours :-). Yes, now it is clearer why we are getting the failure when not having `NoIncludeNullableAttributes`. I thought the nullable attribute definition is not added to the test projects but this is not the case. So, your original change having `NoIncludeNullableAttributes` make sense now.

CC @ericstj if having any better suggestion.

Reply:
Maybe some change could be made to suppress a warning in local definition when using internals visible to.  IMO it's better to have a local definition than depend on one in another assembly.  It's just tests here, so no big deal but wanted to share the suggestion.